### PR TITLE
(#9) Create contracts package

### PR DESCRIPTION
**Pull request by pikonha**
- **chore: init foundry package**
- **test: public resolver read and write**
- **feat: public resolver deployment script**
- **chore: add usage to the readme**

# Create contracts package

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description
This adds the contracts package and all the requirements to either run it locally, deploy to mainnet/testnet or run tests.

## Related Issue
#9 

## Changes
- [X] New feature implementation

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

## Additional Notes
Usage instructions were added to the root README.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```
Comment (pikonha) [inline .env.example:1]: this is a dummy private key generated by [anvil](https://book.getfoundry.sh/reference/anvil/)

File: packages/contracts/lib/forge-std/.github/workflows/ci.yml
Comment (pikonha) [inline packages/contracts/lib/forge-std/.github/workflows/ci.yml:1]: this is a git submodule vendorized by Foundry

File: packages/contracts/package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "contracts",
+  "packageManager": "yarn@4.1.0",
+  "scripts": {
+    "test": "forge test -vv",
+    "test:watch": "forge test -vv --watch",
+    "start": "anvil & copyfiles ../../.env contracts/.env && forge script script/*.sol --rpc-url http://localhost:8545 --broadcast -vv",
```
Comment (pikonha) [inline packages/contracts/package.json:7]: the `.env` copying was required given that Foundry [can't handle env files on a custom path](https://github.com/foundry-rs/foundry/issues/5723)

File: packages/contracts/package.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "contracts",
+  "packageManager": "yarn@4.1.0",
+  "scripts": {
+    "test": "forge test -vv",
+    "test:watch": "forge test -vv --watch",
```
Comment (pikonha) [inline packages/contracts/package.json:6]: this is the command that should be used for development

File: README.md [hunk 4]
```diff
@@ -11,6 +11,25 @@ The primary objective of this project is to enhance the scalability of ENS while
...
+2. Install dependencies by running:
+
+```bash
+yarn install
+```
+
+3. Start the development server:
+
+```bash
+yarn start
```
Comment (LeonardoVieira1630) [inline README.md:28]: Hey Lucas, i was testing the branch and i was not able to run this command (yarn start) that easy. I figure it out that it is necessary to add a .env file in the contracts folder. Maybe, it'll be good to explain this process here in the readme. It will make thinks more clear to someone that is trying to run our project.

File: packages/contracts/lib/forge-std/.github/workflows/ci.yml
Comment (LeonardoVieira1630) [inline packages/contracts/lib/forge-std/.github/workflows/ci.yml:1]: Lucas, is this file necessary?

**Review by LeonardoVieira1630 — CHANGES_REQUESTED** [review]
Hello lucas. I was reviewing the PR and saw some things that i want to discuss with you. First, i let a comment in the readme with a suggestion to improve it. Take a look at it and let me know what you think.
Also, i saw that you commited a /lib folder and ir has a lot of files. Do we really need to have all of it in the repository? Isnt it a solution where it does not need to be commited?

File: packages/contracts/lib/forge-std/.github/workflows/sync.yml
[the remark this answers is not in the document]
Comment (pikonha) [reply packages/contracts/lib/forge-std/.github/workflows/sync.yml:1]: replaced by gitmodule

File: packages/contracts/lib/forge-std/.github/workflows/ci.yml
Comment (pikonha) [reply packages/contracts/lib/forge-std/.github/workflows/ci.yml:1]: replaced by gitmodule

**Review by LeonardoVieira1630 — APPROVED** [review]
